### PR TITLE
HBASE-22664 Move protobuf stuff in hbase-rsgroup to hbase-protocol-sh…

### PR DESCRIPTION
**Pull request by Apache9**
…aded

### CODE REVIEW

File: hbase-client/src/main/java/org/apache/hadoop/hbase/protobuf/ProtobufUtil.java [hunk 1]
```diff
@@ -1768,23 +1768,36 @@ public static ServerName toServerName(final byte [] data) throws Deserialization
...
+    for (HBaseProtos.TableName pTableName : proto.getTablesList()) {
       RSGroupInfo.addTable(ProtobufUtil.toTableName(pTableName));
     }
     return RSGroupInfo;
   }
 
-  public static HBaseProtos.TimeRange toTimeRange(TimeRange timeRange) {
-    if (timeRange == null) {
-      timeRange = TimeRange.allTime();
+  public static RSGroupProtos.RSGroupInfo toProtoGroupInfo(RSGroupInfo pojo) {
```
Comment (infraio) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/protobuf/ProtobufUtil.java]: This method should be removed?

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 23 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 24 | Maven dependency ordering for branch |
| +1 | mvninstall | 245 | master passed |
| +1 | compile | 105 | master passed |
| +1 | checkstyle | 60 | master passed |
| +1 | shadedjars | 268 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 321 | master passed |
| +1 | javadoc | 63 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 16 | Maven dependency ordering for patch |
| +1 | mvninstall | 241 | the patch passed |
| +1 | compile | 104 | the patch passed |
| +1 | cc | 104 | the patch passed |
| +1 | javac | 104 | the patch passed |
| +1 | checkstyle | 60 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 2 | The patch has no ill-formed XML file. |
| +1 | shadedjars | 269 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 761 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | hbaseprotoc | 121 | the patch passed |
| +1 | findbugs | 362 | the patch passed |
| +1 | javadoc | 63 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 37 | hbase-protocol-shaded in the patch passed. |
| +1 | unit | 24 | hbase-protocol in the patch passed. |
| +1 | unit | 109 | hbase-client in the patch passed. |
| +1 | unit | 221 | hbase-rsgroup in the patch passed. |
| +1 | asflicense | 46 | The patch does not generate ASF License warnings. |
| | | 3918 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-362/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/362 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  cc  hbaseprotoc  xml  |
| uname | Linux 3dcf9ec97852 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 605f8a15bb |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-362/1/testReport/ |
| Max. process+thread count | 4602 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-protocol hbase-client hbase-rsgroup U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-362/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 52 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 30 | Maven dependency ordering for branch |
| +1 | mvninstall | 275 | master passed |
| +1 | compile | 103 | master passed |
| +1 | checkstyle | 61 | master passed |
| +1 | shadedjars | 270 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 327 | master passed |
| +1 | javadoc | 68 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 16 | Maven dependency ordering for patch |
| +1 | mvninstall | 237 | the patch passed |
| +1 | compile | 104 | the patch passed |
| +1 | cc | 104 | the patch passed |
| +1 | javac | 104 | the patch passed |
| +1 | checkstyle | 67 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 1 | The patch has no ill-formed XML file. |
| +1 | shadedjars | 274 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 791 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | hbaseprotoc | 122 | the patch passed |
| +1 | findbugs | 353 | the patch passed |
| +1 | javadoc | 60 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 37 | hbase-protocol-shaded in the patch passed. |
| +1 | unit | 25 | hbase-protocol in the patch passed. |
| +1 | unit | 108 | hbase-client in the patch passed. |
| -1 | unit | 574 | hbase-rsgroup in the patch failed. |
| +1 | asflicense | 43 | The patch does not generate ASF License warnings. |
| | | 4386 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.rsgroup.TestRSGroupsBalance |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-362/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/362 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  cc  hbaseprotoc  xml  |
| uname | Linux 5bf55cae4981 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 605f8a15bb |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-362/2/artifact/out/patch-unit-hbase-rsgroup.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-362/2/testReport/ |
| Max. process+thread count | 4514 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-protocol hbase-client hbase-rsgroup U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-362/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/protobuf/ProtobufUtil.java (hunk 1 above)
Comment (Apache9) [reply hbase-client/src/main/java/org/apache/hadoop/hbase/protobuf/ProtobufUtil.java]: At least we still need it for now as it is referenced in lots of places. Not sure if we can remove it at the end. FWIW, I think we still need to support the old way to change rsgroup, which uses coprocessor endpoint?

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 151 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 25 | Maven dependency ordering for branch |
| +1 | mvninstall | 252 | master passed |
| +1 | compile | 106 | master passed |
| +1 | checkstyle | 69 | master passed |
| +1 | shadedjars | 266 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 321 | master passed |
| +1 | javadoc | 61 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 13 | Maven dependency ordering for patch |
| +1 | mvninstall | 240 | the patch passed |
| +1 | compile | 107 | the patch passed |
| +1 | cc | 107 | the patch passed |
| +1 | javac | 107 | the patch passed |
| +1 | checkstyle | 65 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 2 | The patch has no ill-formed XML file. |
| +1 | shadedjars | 269 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 746 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | hbaseprotoc | 108 | the patch passed |
| +1 | findbugs | 399 | the patch passed |
| +1 | javadoc | 57 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 36 | hbase-protocol-shaded in the patch passed. |
| +1 | unit | 25 | hbase-protocol in the patch passed. |
| +1 | unit | 111 | hbase-client in the patch passed. |
| +1 | unit | 463 | hbase-rsgroup in the patch passed. |
| +1 | asflicense | 46 | The patch does not generate ASF License warnings. |
| | | 4298 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-362/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/362 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  cc  hbaseprotoc  xml  |
| uname | Linux 6270fc02cad9 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 605f8a15bb |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-362/3/testReport/ |
| Max. process+thread count | 4410 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-protocol hbase-client hbase-rsgroup U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-362/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
Any other concerns? @infraio 
Thanks.

**Pull request merged**